File: m/rls/rlb/stidx.c
/* file -  stidx - character index in string */
#include "m:\rid\rider.h"
#include "m:\rid\stdef.h"
/* code -  st_idx - character index in string */
int st_idx(
int cha ,
char *str )
{ int idx = 0;
  int tmp ;
  if( str == NULL)return ( -1 );
  while ( (tmp = *str++) != 0) {
    if( ((BYTE )tmp == (BYTE )cha))return ( idx );
    ++idx;
  } 
  return ( -1);
} 

File: m/rls/stidx.r
file	stidx - character index in string
include	rid:rider
include	rid:stdef

code	st_idx - character index in string

  func	st_idx
	cha : int			; the non-zero character
	str : * char			; the set (or <> for empty set)
	()  : int			; -1 => no set or no match
					; +n => index of located member
  is	idx : int = 0			;
	tmp : int			; avoid double access
	reply -1 if str eq <>		; no set 
	while (tmp = *str++) ne		; no game
	    <BYTE>tmp eq <BYTE>cha	; match
	   reply idx if that		; game, set and match
	   ++idx			;
	end				;
	reply -1			;
  end
